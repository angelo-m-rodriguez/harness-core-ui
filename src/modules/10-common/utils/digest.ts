/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

async function digest(str: string): Promise<string> {
  const msgUint8 = new TextEncoder().encode(str) // encode as (utf-8) Uint8Array
  const hashBuffer = await crypto.subtle.digest('SHA-256', msgUint8) // hash the message
  const hashArray = Array.from(new Uint8Array(hashBuffer)) // convert buffer to byte array
  const hashHex = hashArray.map(b => b.toString(16).padStart(2, '0')).join('') // convert bytes to hex string

  return hashHex
}

export default digest
