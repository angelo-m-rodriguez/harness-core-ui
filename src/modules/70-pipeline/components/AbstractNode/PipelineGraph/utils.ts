import type { RefObject } from 'react'
import type { StageElementWrapperConfig } from 'services/cd-ng'
const INITIAL_ZOOM_LEVEL = 1
const ZOOM_INC_DEC_LEVEL = 0.1
const getFinalSVGArrowPath = (parentId: string, id1 = '', id2 = '', isParallelNode = false): string => {
  if (!parentId) {
    return ''
  }
  const node1 = getComputedPosition(parentId, id1)
  const node2 = getComputedPosition(parentId, id2)
  if (!node1 || !node2) {
    return ''
  }

  const node1VerticalMid = node1.top + node1.height / 2
  const node2VerticalMid = node2.top + node2.height / 2

  const startPoint = `${node1.right},${node1VerticalMid}`
  const horizontalMid = Math.abs((node1.right + node2.left) / 2)
  const endPoint = `${node2.left},${node2VerticalMid}`

  if (node2.y < node1.y) {
    //  child node is at top
    const curveLeftToTop = `Q${horizontalMid},${node1VerticalMid} ${horizontalMid},${node1VerticalMid - 20}`
    const curveBottomToRight = `Q${horizontalMid},${node2VerticalMid} ${horizontalMid + 20},${node2VerticalMid}`
    return `M${startPoint} L${horizontalMid - 20},${node1VerticalMid} ${curveLeftToTop} 
    L${horizontalMid},${node2VerticalMid + 20} ${curveBottomToRight} L${endPoint}`
  } else if (node1.y === node2.y) {
    // both nodes are at same level vertically
    return `M${startPoint}  L${endPoint} `
  } else {
    //  child node is at bottom
    const curveLeftToBottom = `Q${horizontalMid},${node1VerticalMid} ${horizontalMid},${node1VerticalMid + 20}`

    const curveTopToRight = `Q${horizontalMid},${node2VerticalMid} ${horizontalMid + 20},${node2VerticalMid}`

    if (isParallelNode) {
      const updatedStart = node1.left - 45 // new start point
      const parallelLinkStart = `${
        updatedStart // half of link length
      },${node1VerticalMid}`

      const curveLBparallel = `Q${updatedStart + 20},${node1VerticalMid} ${updatedStart + 20},${node1VerticalMid + 20} `
      const curveTRparallel = `Q${updatedStart + 20},${node2VerticalMid} ${updatedStart + 40},${node2VerticalMid}`

      return `M${parallelLinkStart} 
      ${curveLBparallel} 
      L${updatedStart + 20},${node2VerticalMid - 20} 
      ${curveTRparallel} 
      L${node2.left},${node2VerticalMid} 
      M${node2.right},${node2VerticalMid} 
      L${node2.right + 10},${node2VerticalMid}
      Q${node2.right + 25},${node2VerticalMid} ${node2.right + 25},${node2VerticalMid - 20}      
      L${node2.right + 25},${node1VerticalMid + 20}
      Q${node2.right + 25},${node1VerticalMid} ${node2.right + 40},${node1VerticalMid}
      `
    }
    return `M${startPoint} L${horizontalMid - 20},${node1VerticalMid} ${curveLeftToBottom} 
    L${horizontalMid},${node2VerticalMid - 20} ${curveTopToRight} L${endPoint}`
  }
}

const getComputedPosition = (parentId: string, childId: string): DOMRect => {
  const parentPos = document.getElementById(parentId)?.getBoundingClientRect() as DOMRect
  const childPos = document.getElementById(childId)?.getBoundingClientRect() as DOMRect
  const updatedTop = childPos.top - parentPos.top
  const updatedLeft = childPos.left - parentPos.left
  const updatedRight = updatedLeft + childPos.width
  const updatedBottom = updatedTop + childPos.height
  const updatedPos: DOMRect = {
    ...childPos,
    left: updatedLeft,
    top: updatedTop,
    right: updatedRight,
    bottom: updatedBottom,
    width: childPos.width,
    height: childPos.height,
    x: childPos.x,
    y: childPos.y
  }
  return updatedPos
}

const setupDragEventListeners = (canvasRef: RefObject<HTMLDivElement>): (() => void) => {
  let offset = [0, 0]
  const divOverlay = (canvasRef.current as HTMLDivElement).parentElement as HTMLDivElement
  const canvas = canvasRef.current as HTMLDivElement
  let isDown = false

  const onMouseMove = (e: any): void => {
    e.preventDefault()
    if (isDown) {
      canvas.style.left = e.clientX + offset[0] + 'px'
      canvas.style.top = e.clientY + offset[1] + 'px'
    }
  }

  const onMouseDown = (e: any): void => {
    isDown = true
    offset = [canvas.offsetLeft - e.clientX, canvas.offsetTop - e.clientY]
  }
  const onMouseUp = (): void => {
    isDown = false
  }

  divOverlay.addEventListener('mousedown', onMouseDown, true)
  document.addEventListener('mouseup', onMouseUp, true)
  document.addEventListener('mousemove', onMouseMove, true)
  return () => {
    divOverlay.removeEventListener('mousedown', onMouseDown)
    document.removeEventListener('mouseup', onMouseUp)
    document.removeEventListener('mousemove', onMouseMove)
  }
}

const getSVGLinksFromPipeline = (stages?: StageElementWrapperConfig[], resultArr: string[] = []): string[] => {
  let prevElement: StageElementWrapperConfig
  stages?.forEach(stage => {
    if (!prevElement) {
      prevElement = stage
      return
    }
    if (stage.parallel) {
      getParallelNodeLinks(stage.parallel, prevElement, resultArr)
    } else {
      resultArr.push(
        getFinalSVGArrowPath('tree-container', prevElement.stage?.identifier as string, stage.stage?.identifier, false)
      )
    }
    prevElement = stage
  })
  return resultArr
}
const getParallelNodeLinks = (
  stages: StageElementWrapperConfig[],
  prevStage: StageElementWrapperConfig,
  resultArr: string[] = []
): void => {
  const [firstStage, ...restStages] = stages
  resultArr.push(
    getFinalSVGArrowPath('tree-container', prevStage.stage?.identifier as string, firstStage.stage?.identifier, false)
  )
  restStages?.forEach(stage => {
    resultArr.push(
      getFinalSVGArrowPath('tree-container', firstStage.stage?.identifier as string, stage.stage?.identifier, true)
    )
  })
}

const getScaleToFitValue = (elm: HTMLElement, paddingFromBottom = 20): number => {
  return (
    1 /
    Math.max(
      elm.clientWidth / window.innerWidth,
      elm.clientHeight / (window.innerHeight - elm.offsetTop - paddingFromBottom)
    )
  )
}
export {
  getFinalSVGArrowPath,
  setupDragEventListeners,
  getSVGLinksFromPipeline,
  ZOOM_INC_DEC_LEVEL,
  INITIAL_ZOOM_LEVEL,
  getScaleToFitValue
}
