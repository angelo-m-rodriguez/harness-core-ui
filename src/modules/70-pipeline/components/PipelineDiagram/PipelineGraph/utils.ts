import { defaultTo } from 'lodash-es'
import type { PipelineGraphState } from '../types'

const shouldAttachRef = (
  intersectingIndex: number,
  isCurrentChildLast: boolean,
  indexRelativeToParent: number
): boolean => {
  return intersectingIndex === -1 // intersecting index is not set
    ? isCurrentChildLast // current child is not the last child in parallel
    : intersectingIndex === indexRelativeToParent // intersecting index is not equal to indexRelativeToParent (i.e index+1 as parent is 0)
}

const shouldRenderGroupNode = (attachRef: boolean, isCurrentChildLast: boolean): boolean => {
  return attachRef && !isCurrentChildLast // render group node is ref is attached and current node is not the last child
}

const isFirstNodeAGroupNode = (intersectingIndex: number, collapseOnIntersect = false, childLength = 0): boolean =>
  intersectingIndex === 0 && collapseOnIntersect && childLength > 0 // if intersecting index is 0 and collapseOnIntersect is enabled

const showChildNode = (indexRelativeToParent: number, intersectingIndex: number): boolean => {
  return !(indexRelativeToParent > intersectingIndex && intersectingIndex !== -1) // child index is greater than intersectingIndex and intersectingIndex is set to some value
}

const isNodeParallel = (node: PipelineGraphState): boolean => defaultTo(node?.children?.length, 0) > 0

export { shouldAttachRef, shouldRenderGroupNode, isFirstNodeAGroupNode, showChildNode, isNodeParallel }
