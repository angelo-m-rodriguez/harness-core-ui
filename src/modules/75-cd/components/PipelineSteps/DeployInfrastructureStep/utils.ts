/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import type { FormikProps } from 'formik'
import { isEmpty } from 'lodash-es'

import { getMultiTypeFromValue, MultiTypeInputType, SelectOption } from '@harness/uicore'
import type { EnvironmentResponseDTO, PipelineInfrastructure } from 'services/cd-ng'

import type { StepViewType } from '@pipeline/components/AbstractSteps/Step'

export interface PipelineInfrastructureV2 extends PipelineInfrastructure {
  environmentOrEnvGroupRef?: SelectOption
  environmentGroup?: any
  environmentRef2?: any
  infrastructureRef?: any
}

export interface DeployInfrastructureProps {
  initialValues: PipelineInfrastructureV2
  onUpdate?: (data: PipelineInfrastructureV2) => void
  readonly: boolean
  allowableTypes: MultiTypeInputType[]
  stepViewType?: StepViewType
  inputSetData?: {
    template?: PipelineInfrastructureV2
    path?: string
    readonly?: boolean
  }
}

export interface DeployInfrastructureState {
  isEdit: boolean
  isEnvironment: boolean
  formik?: FormikProps<PipelineInfrastructureV2>
  data?: EnvironmentResponseDTO
}

export function isEditEnvironment(data: PipelineInfrastructureV2): boolean {
  if (
    getMultiTypeFromValue(data.environmentRef) !== MultiTypeInputType.RUNTIME &&
    !isEmpty(data.environmentOrEnvGroupRef)
  ) {
    return true
  }
  return false
}

export function isEditInfrastructure(data: any): boolean {
  if (
    getMultiTypeFromValue(data.infrastructureRef) !== MultiTypeInputType.RUNTIME &&
    !isEmpty(data.infrastructureRef)
  ) {
    return true
  } else if (data.infrastructure && !isEmpty(data.infrastructure.identifier)) {
    return true
  }
  return false
}
