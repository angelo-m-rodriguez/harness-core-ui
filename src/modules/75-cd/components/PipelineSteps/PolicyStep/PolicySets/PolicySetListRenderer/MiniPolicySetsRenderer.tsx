/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React from 'react'
import { useParams } from 'react-router-dom'
import { defaultTo } from 'lodash-es'
import { Spinner } from '@blueprintjs/core'

import { Color, Layout, Text, useToaster } from '@harness/uicore'
import { GetPolicySet, GetPolicySetQueryParams, LinkedPolicy } from 'services/pm'

import type { ProjectPathProps } from '@common/interfaces/RouteInterfaces'

import { PolicySetType } from '../../BasePolicyStep'
import { getErrorMessage } from '../PolicySetsFormField/PolicySetsFormField'

import css from './PolicySetListRenderer.module.scss'

interface MiniPolicySetsRendererProps {
  policySetIds: string[]
}

export function MiniPolicySetsRenderer({ policySetIds }: MiniPolicySetsRendererProps) {
  const { accountId: accountIdentifier, projectIdentifier, orgIdentifier } = useParams<ProjectPathProps>()
  const { showError } = useToaster()
  return (
    <>
      {policySetIds.map(policySetId => {
        const policySetType = policySetId.includes('acc.')
          ? PolicySetType.ACCOUNT
          : policySetId.includes('org.')
          ? PolicySetType.ORG
          : PolicySetType.PROJECT

        const queryParams: GetPolicySetQueryParams = {
          accountIdentifier,
          ...((policySetType === PolicySetType.ORG || policySetType === PolicySetType.PROJECT) && { orgIdentifier }),
          ...(policySetType === PolicySetType.PROJECT && { projectIdentifier })
        }

        return (
          <GetPolicySet policyset={policySetId} queryParams={{ ...queryParams }} key={policySetId}>
            {(policySet, { loading, error }) => {
              if (error) {
                showError(getErrorMessage(error))
                return null
              } else {
                return (
                  <>
                    {policySet && (
                      <Layout.Horizontal
                        className={css.policySetHolder}
                        flex={{ justifyContent: 'space-between', alignItems: 'center' }}
                      >
                        {loading ? (
                          <Spinner size={Spinner.SIZE_SMALL} />
                        ) : (
                          <>
                            <Text lineClamp={1} color={Color.BLACK}>
                              {policySet.name}
                            </Text>
                            <Layout.Horizontal flex={{ justifyContent: 'flex-end', alignItems: 'center' }}>
                              <MiniPoliciesRenderer policies={defaultTo(policySet.policies, [])} />
                              <Text font={'small'} width={48}>
                                {policySetType}
                              </Text>
                            </Layout.Horizontal>
                          </>
                        )}
                      </Layout.Horizontal>
                    )}
                  </>
                )
              }
            }}
          </GetPolicySet>
        )
      })}
    </>
  )
}

interface MiniPoliciesRendererProps {
  policies: LinkedPolicy[]
}

export const MiniPoliciesRenderer = ({ policies }: MiniPoliciesRendererProps) => {
  const length = policies.length
  // istanbul ignore else
  if (length === 0) {
    return null
  }
  const policyNames = policies.map(policy => policy.name)

  return (
    <Layout.Horizontal flex={{ alignItems: 'center' }} margin={{ right: 'small' }}>
      {policyNames.slice(0, 2).map((policy, index) => (
        <Text className={css.styledPolicy} key={index} lineClamp={1}>
          {policy}
        </Text>
      ))}
      {length > 2 && (
        <Text
          className={css.styledPolicy}
          background={Color.GREY_100}
          alwaysShowTooltip
          tooltip={
            <Layout.Vertical padding="medium">
              {policyNames.splice(2).map((policy, index) => (
                <Text
                  lineClamp={1}
                  color={Color.BLACK}
                  margin={{ top: 'small', bottom: 'small' }}
                  style={{ maxWidth: '400px' }}
                  key={index}
                >
                  {policy}
                </Text>
              ))}
            </Layout.Vertical>
          }
        >
          {`+${length - 2}`}
        </Text>
      )}
    </Layout.Horizontal>
  )
}
