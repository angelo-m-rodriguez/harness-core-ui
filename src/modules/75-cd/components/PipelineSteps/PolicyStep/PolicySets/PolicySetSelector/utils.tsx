/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React from 'react'
import { FontVariation, Layout, Text } from '@harness/uicore'
import { useStrings } from 'framework/strings'

export const PolicySetSelectorDialogTitle = (): JSX.Element => {
  const { getString } = useStrings()
  return (
    <Layout.Horizontal
      spacing="xsmall"
      padding={{ top: 'xxlarge', left: 'medium', right: 'large', bottom: 'medium' }}
      flex={{ justifyContent: 'space-between' }}
    >
      <Layout.Vertical>
        <Text font={{ variation: FontVariation.H3 }}>{getString('common.policiesSets.selectPolicySet')}</Text>
      </Layout.Vertical>
    </Layout.Horizontal>
  )
}
