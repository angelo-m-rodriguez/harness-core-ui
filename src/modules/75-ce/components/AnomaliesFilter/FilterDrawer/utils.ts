/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import type { MultiSelectOption } from '@harness/uicore'
import { omit } from 'lodash-es'
import type { AnomalyFilterProperties } from 'services/ce'
import type { AnomaliesFilterFormType } from './FilterDrawer'

const getMultiSelectOptions = (values: string[]): MultiSelectOption[] =>
  values.map(item => ({ label: item, value: item }))

const getValueFromOption = (values: MultiSelectOption[]): string[] => values.map(item => item.value.toString())

export const getAnomalyFormValuesFromFilterProperties = (
  filterProperties: AnomalyFilterProperties
): AnomaliesFilterFormType => {
  const formValues: AnomaliesFilterFormType = {}

  Object.entries(omit(filterProperties, 'filterType')).forEach(([key, value]) => {
    if (value) {
      if (key === 'minActualAmount' || key === 'minAnomalousSpend') {
        formValues[key] = +value
      } else {
        ;(formValues as any)[key] = getMultiSelectOptions(value as string[])
      }
    }
  })

  return formValues
}

export const getAnomalyFilterPropertiesFromForm = (formData: AnomaliesFilterFormType): AnomalyFilterProperties => {
  const filterProperties: AnomalyFilterProperties = { filterType: 'Anomaly' }

  Object.entries(formData).forEach(([key, value]) => {
    if (key === 'minActualAmount' || key === 'minAnomalousSpend') {
      filterProperties[key] = +value
    } else {
      ;(filterProperties as any)[key] = getValueFromOption(value)
    }
  })

  return filterProperties
}
