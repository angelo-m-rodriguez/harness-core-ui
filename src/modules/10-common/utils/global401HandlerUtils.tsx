/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import type { History } from 'history'
import routes from '@common/RouteDefinitions'
import SecureStorage from 'framework/utils/SecureStorage'
import { getLoginPageURL } from 'framework/utils/SessionUtils'
import { returnUrlParams } from './routeUtils'

export const global401HandlerUtils = (history: History) => {
  SecureStorage.clear()
  history.push({
    pathname: routes.toRedirect(),
    search: returnUrlParams(getLoginPageURL({ returnUrl: window.location.href }))
  })
}
