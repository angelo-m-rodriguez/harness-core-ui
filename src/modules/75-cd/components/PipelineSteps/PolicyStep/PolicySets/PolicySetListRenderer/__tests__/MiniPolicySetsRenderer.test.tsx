/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React from 'react'
import { render } from '@testing-library/react'

import type { PolicySetWithLinkedPolicies } from 'services/pm'

import { TestWrapper } from '@common/utils/testUtils'

import { MiniPolicySetsRenderer } from '../MiniPolicySetsRenderer'

jest.mock('@common/exports', () => ({
  useToaster: () => ({
    showError: jest.fn()
  })
}))

jest.mock('services/pm', () => ({
  GetPolicySet: jest.fn().mockImplementation((props: any) => {
    const MockComponent = () => {
      let policySet: PolicySetWithLinkedPolicies = {}
      if (props.policyset === 'acc.test') {
        policySet = {
          name: 'Policy set 1',
          policies: [
            {
              name: 'Policy 1'
            },
            {
              name: 'Policy 2'
            }
          ]
        }
      } else if (props.policyset === 'org.test') {
        policySet = {
          name: 'policy set 2',
          policies: [
            {
              name: 'policy 3'
            },
            {
              name: 'policy 4'
            },
            {
              name: 'policy 5'
            }
          ]
        }
      } else {
        policySet = {
          name: 'policy set 3',
          policies: []
        }
      }
      return props.children(policySet, {
        loading: props.policyset === 'loadingId',
        error: false
      })
    }

    return <MockComponent {...props} />
  })
}))

describe('Mini Policy Sets Renderer', () => {
  test('snapshot test', () => {
    const { container } = render(
      <TestWrapper
        queryParams={{
          accountIdentifier: 'acc',
          orgIdentifier: 'org',
          projectIdentifier: 'project'
        }}
      >
        <MiniPolicySetsRenderer policySetIds={['acc.test', 'org.test', 'test', 'loadingId']} />
      </TestWrapper>
    )

    expect(container).toMatchSnapshot()
  })
})
