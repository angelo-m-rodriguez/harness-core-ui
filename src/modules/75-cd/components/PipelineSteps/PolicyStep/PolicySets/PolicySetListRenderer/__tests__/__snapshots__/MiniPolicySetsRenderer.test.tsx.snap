// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mini Policy Sets Renderer snapshot test 1`] = `
<div>
  <div
    class="Layout--horizontal StyledProps--main policySetHolder StyledProps--flex StyledProps--flex-justifyContent-space-between StyledProps--flex-alignItems-center"
  >
    <p
      class="StyledProps--font Text--lineclamp StyledProps--main StyledProps--color StyledProps--color-black"
      style="--text-line-clamp: 1;"
    >
      Policy set 1
    </p>
    <div
      class="Layout--horizontal StyledProps--main StyledProps--flex StyledProps--flex-justifyContent-flex-end StyledProps--flex-alignItems-center"
    >
      <div
        class="Layout--horizontal StyledProps--main StyledProps--flex StyledProps--flex-alignItems-center StyledProps--margin-right-small"
      >
        <p
          class="StyledProps--font Text--lineclamp StyledProps--main styledPolicy"
          style="--text-line-clamp: 1;"
        >
          Policy 1
        </p>
        <p
          class="StyledProps--font Text--lineclamp StyledProps--main styledPolicy"
          style="--text-line-clamp: 1;"
        >
          Policy 2
        </p>
      </div>
      <p
        class="StyledProps--font StyledProps--main StyledProps--font-small"
        style="width: 48px;"
      >
        Account
      </p>
    </div>
  </div>
  <div
    class="Layout--horizontal StyledProps--main policySetHolder StyledProps--flex StyledProps--flex-justifyContent-space-between StyledProps--flex-alignItems-center"
  >
    <p
      class="StyledProps--font Text--lineclamp StyledProps--main StyledProps--color StyledProps--color-black"
      style="--text-line-clamp: 1;"
    >
      policy set 2
    </p>
    <div
      class="Layout--horizontal StyledProps--main StyledProps--flex StyledProps--flex-justifyContent-flex-end StyledProps--flex-alignItems-center"
    >
      <div
        class="Layout--horizontal StyledProps--main StyledProps--flex StyledProps--flex-alignItems-center StyledProps--margin-right-small"
      >
        <p
          class="StyledProps--font Text--lineclamp StyledProps--main styledPolicy"
          style="--text-line-clamp: 1;"
        >
          policy 3
        </p>
        <p
          class="StyledProps--font Text--lineclamp StyledProps--main styledPolicy"
          style="--text-line-clamp: 1;"
        >
          policy 4
        </p>
        <span
          class="bp3-popover-wrapper"
        >
          <span
            class="bp3-popover-target Text--targetClass"
          >
            <p
              class="StyledProps--font StyledProps--main styledPolicy StyledProps--background-grey100"
              tabindex="0"
            >
              +1
            </p>
          </span>
        </span>
      </div>
      <p
        class="StyledProps--font StyledProps--main StyledProps--font-small"
        style="width: 48px;"
      >
        Org
      </p>
    </div>
  </div>
  <div
    class="Layout--horizontal StyledProps--main policySetHolder StyledProps--flex StyledProps--flex-justifyContent-space-between StyledProps--flex-alignItems-center"
  >
    <p
      class="StyledProps--font Text--lineclamp StyledProps--main StyledProps--color StyledProps--color-black"
      style="--text-line-clamp: 1;"
    >
      policy set 3
    </p>
    <div
      class="Layout--horizontal StyledProps--main StyledProps--flex StyledProps--flex-justifyContent-flex-end StyledProps--flex-alignItems-center"
    >
      <p
        class="StyledProps--font StyledProps--main StyledProps--font-small"
        style="width: 48px;"
      >
        Project
      </p>
    </div>
  </div>
  <div
    class="Layout--horizontal StyledProps--main policySetHolder StyledProps--flex StyledProps--flex-justifyContent-space-between StyledProps--flex-alignItems-center"
  >
    <div
      class="bp3-spinner"
    >
      <div
        class="bp3-spinner-animation"
      >
        <svg
          height="20"
          stroke-width="16.00"
          viewBox="-3.00 -3.00 106.00 106.00"
          width="20"
        >
          <path
            class="bp3-spinner-track"
            d="M 50,50 m 0,-45 a 45,45 0 1 1 0,90 a 45,45 0 1 1 0,-90"
          />
          <path
            class="bp3-spinner-head"
            d="M 50,50 m 0,-45 a 45,45 0 1 1 0,90 a 45,45 0 1 1 0,-90"
            pathLength="280"
            stroke-dasharray="280 280"
            stroke-dashoffset="210"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;
